risc0/zkvm/src/exec/mod.rs

```diff
@@ -125,6 +126,17 @@ impl Write for Journal {
 impl<'a> Executor<'a> {
     /// Construct a new [Executor] from a [MemoryImage] and entry point.
     pub fn new(env: ExecutorEnv<'a>, image: MemoryImage, pc: u32) -> Self {
+        Self::new_with_id(env, image, pc, None)
+    }
+
+    /// Construct a new [Executor] from a [MemoryImage], entry point, along with
+    /// the proof ID generated from bonsai
+    fn new_with_id(
```

Comment:
I know @flaub and I chatted about making a Executor trait like interface, similar to how the Prover pluggable system works. I think this is a good use case for that new API.